src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -132,6 +132,37 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
                     throw new FormatException("The assembly file version is invalid: " + fileVersion, ex);
                 }
             }
+            
+            var analyzerOptionsJson = rawProject.Value("analyzerOptions") as JsonObject;
+            if (analyzerOptionsJson != null)
+            {
+                var analyzerOptions = new AnalyzerOptions();
+                
+                foreach (var key in analyzerOptionsJson.Keys)
+                {
+                    switch (key)
+                    {
+                        case "languageId":
```

Comment:
Just talked with @srivatsn offline -- he had a compelling argument for inference beyond just analyzers. I'll add it.